libmamba/src/core/solver.cpp

```diff
@@ -147,53 +152,73 @@ namespace mamba
         }
     }
 
-    std::string MSolverProblem::to_string() const
+    namespace
     {
-        return solver_problemruleinfo2str(
-            solver, (SolverRuleinfo) type, source_id, target_id, dep_id);
-    }
+        std::optional<PackageInfo> make_solver_problem_source(::Pool* pool, Id source_id)
+        {
+            if (source_id == 0 || source_id >= pool->nsolvables)
+            {
+                return std::nullopt;
+            }
+            return pool_id2solvable(pool, source_id);
+        }
 
-    std::optional<PackageInfo> MSolverProblem::target() const
-    {
-        if (target_id == 0 || target_id >= solver->pool->nsolvables)
-            return std::nullopt;
-        return pool_id2solvable(solver->pool, target_id);
-    }
+        std::optional<PackageInfo> make_solver_problem_target(::Pool* pool, Id target_id)
+        {
+            if (target_id == 0 || target_id >= pool->nsolvables)
+            {
+                return std::nullopt;
+            }
+            return pool_id2solvable(pool, target_id);
+        }
 
-    std::optional<PackageInfo> MSolverProblem::source() const
-    {
-        if (source_id == 0 || source_id >= solver->pool->nsolvables)
-            return std::nullopt;
-        ;
-        return pool_id2solvable(solver->pool, source_id);
-    }
 
-    std::optional<std::string> MSolverProblem::dep() const
-    {
-        if (!dep_id)
-            return std::nullopt;
-        return pool_dep2str(solver->pool, dep_id);
-    }
+        std::optional<std::string> make_solver_problem_dep(::Pool* pool, Id dep_id)
+        {
+            if (!dep_id)
+            {
+                return std::nullopt;
+            }
+            return pool_dep2str(pool, dep_id);
+        }
 
-    MSolver::MSolver(MPool& pool, const std::vector<std::pair<int, int>>& flags)
-        : m_flags(flags)
-        , m_is_solved(false)
-        , m_solver(nullptr)
-        , m_pool(pool)
-    {
-        queue_init(&m_jobs);
-        pool_createwhatprovides(pool);
+        MSolverProblem make_solver_problem(
+            ::Solver* solver, SolverRuleinfo type, Id source_id, Id target_id, Id dep_id)
+        {
+            return {
+                /* type= */ type,
```

Comment:
When we have these comments for naming fields, I like to use the C++20 syntax so that we will just uncomment and it will work, for example here: `/* .type = */ type,` (add a dot)